### ISSUE TITLE
Upgrade taking a long time

### ISSUE THREAD

**Issue by MLWALK3R** [owncloud/core#14279]
owncloud/core/issues/14156 @DeepDiver1975
### Steps to reproduce
1. occ upgrade
2. wait

### Expected behaviour
Upgrade should take less than 1HR

### Actual behaviour
Upgrade takes several hours

### Server configuration
**Operating system**: Ubuntu 14.04.1 LTS

**Web server:** nginx version: nginx/1.6.2

**Database:**  Ver 15.1 Distrib 10.1.2-MariaDB

**PHP version:** PHP 5.5.9-1ubuntu4.5 ( OPcache v7.0.3)

**ownCloud version:** (see ownCloud admin page) 8.0 now

**Updated from an older ownCloud or fresh install:** 7.* to 8.0

**List of activated apps:** https://p.ftwnet.co.uk/1424165634

**The content of config/config.php:** https://p.ftwnet.co.uk/1424165739

**Are you using external storage, if yes which one:** Local Disk

**Are you using encryption:** yes

### Client configuration
**Browser:** Chrome

**Operating system:** Win 8.1

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
https://p.ftwnet.co.uk/1424166434 (last 100 lines)
```

**Comment by PVince81** [comment]
Do you have a lot of files, shares or users ?

**Comment by MLWALK3R** [comment]
1 User account, 37.8 GB of files.

**Comment by PVince81** [comment]
Ok, I see you are using encryption. In OC 8 the structure of the encryption keys has changed, so it is likely that what took a long time was the encryption key migration.

CC @schiesbn

**Comment by RealRancor** [comment]
Ref for a similar report: https://github.com/owncloud/core/issues/14012 where @schiesbn has provided some more infos here: https://github.com/owncloud/core/issues/14012#issuecomment-73610374

**Comment by LukasReschke** [comment]
More reports at https://forum.owncloud.org/viewtopic.php?f=31&t=26612:

> I'm running an owncloud instance on a quite decent server (see signature). I got ~26.000 files in my owncloud and encryption enabled for all of them.
> I started the upgrade from the command line yesterday and it's been running now for almost 24 hours. If I check for last changed files I see that it's doing something with the encryption files.

**Comment by PVince81** [comment]
Closing as duplicate of https://github.com/owncloud/core/issues/14012

A fix will be ready for 8.0.2
